crates/sqlexec/src/functions.rs

```diff
@@ -1,206 +1,205 @@
+    /// ```sql
+    /// select user();
+    /// ```
+    User,
+    /// SQL function `current_schema`
+    ///
+    /// `current_schema()` -> `String`
+    /// ```sql
+    /// select current_schema();
+    /// ```
+    CurrentSchema,
+    /// SQL function `current_database`
+    ///
+    /// `current_database()` -> `String`
+    /// ```sql
+    /// select current_database();
+    /// ```
+    CurrentDatabase,
+    /// SQL function `current_catalog`
+    ///
+    /// `current_catalog()` -> `String`
+    /// ```sql
+    /// select current_catalog();
+    /// ```
+    CurrentCatalog,
+    /// SQL function `current_schemas`
+    ///
+    /// `current_schemas()` -> `String[]`
+    ///  current_schemas (include_implicit boolean) -> String[]
     ///
     /// (Postgres)
     /// Get a list of schemas in the current search path.
     CurrentSchemas,
+    Pg(BuiltinPostgresFunctions),
 }
 
-impl BuiltinScalarFunction {
-    /// Try to get the built-in scalar function from the name.
-    pub fn try_from_name(name: &str) -> Option<BuiltinScalarFunction> {
-        // TODO: We can probably move to some fancier function resolution in the
-        // future.
-        Some(match name {
-            "connection_id" => BuiltinScalarFunction::ConnectionId,
-
-            // Postgres system functions.
-            "pg_catalog.current_schemas" => BuiltinScalarFunction::CurrentSchemas,
+#[derive(Debug, Copy, Clone)]
+#[cfg_attr(test, derive(PartialEq))]
+pub enum BuiltinPostgresFunctions {
+    GetUserById,
+    TableIsVisible,
+    EncodingToChar,
+    ArrayToString,
+    HasSchemaPrivilege,
+    HasDatabasePrivilege,
+    HasTablePrivilege,
+}
 
-            // Always fall back to trying to bare pg functions. Longer term will
-            // want to ensure functions are scoped to schemas and do proper
-            // search path resolution.
-            _ => return Self::try_from_name_implicit_pg_catalog(name),
-        })
+impl BuiltinPostgresFunctions {
+    fn into_expr(self, args: Vec<Expr>) -> Expr {
+        let udf = match self {
+            Self::GetUserById => pg_get_userbyid(),
+            Self::TableIsVisible => pg_table_is_visible(),
+            Self::EncodingToChar => pg_encoding_to_char(),
+            Self::ArrayToString => pg_array_to_string(),
+            Self::HasSchemaPrivilege => pg_has_schema_privilege(),
+            Self::HasDatabasePrivilege => pg_has_database_privilege(),
+            Self::HasTablePrivilege => pg_has_table_privilege(),
+        };
+        Expr::ScalarUDF(datafusion::logical_expr::expr::ScalarUDF::new(
+            udf.into(),
+            args,
+        ))
     }
+}
 
-    fn try_from_name_implicit_pg_catalog(name: &str) -> Option<BuiltinScalarFunction> {
-        match name {
-            "current_schemas" => Some(BuiltinScalarFunction::CurrentSchemas),
-            _ => None,
-        }
+impl BuiltinScalarFunction {
+    pub fn find_function(name: &str) -> Option<Self> {
+        Self::from_str(name).ok()
```

Comment:
I believe this is causing a test failure where we can't use the full qualified name (`pg_catalog.current_schemas`). Previously we just split on '.' to get the schema, and I'd be good with that here too.

Reply:
Oh nvm, these should be in the pg funcs:

```
    /// SQL function `current_user`
    ///
    /// `current_user()` -> `String`
    /// ```sql
    /// select current_user();
    /// ```
    CurrentUser,
    /// SQL function `current_role`
    ///
    /// `current_role()` -> `String`
    /// ```sql
    /// select current_role();
    /// ```
    CurrentRole,
    /// SQL function `user`
    ///
    /// `user()` -> `String`
    /// ```sql
    /// select user();
    /// ```
    User,
    /// SQL function `current_schema`
    ///
    /// `current_schema()` -> `String`
    /// ```sql
    /// select current_schema();
    /// ```
    CurrentSchema,
    /// SQL function `current_database`
    ///
    /// `current_database()` -> `String`
    /// ```sql
    /// select current_database();
    /// ```
    CurrentDatabase,
    /// SQL function `current_catalog`
    ///
    /// `current_catalog()` -> `String`
    /// ```sql
    /// select current_catalog();
    /// ```
    CurrentCatalog,
```

Reply:
Ok I think i got all of the pg specific ones moved into `BuiltinPostgresFunctions`